### PR TITLE
[ng] glossary in markdown

### PR DESCRIPTION
**Pull request by chalin**
Contributes to #768

Staged at https://angulardart-org-dev.firebaseapp.com/angular/glossary

### CODE REVIEW

File: src/angular/glossary.md [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+  Ahead-of-time compiled applications also benefit from decreased load time and increased performance.
+</div>
+
+## Annotation
+<div class="l-sub-section" markdown="1">
+  When unqualified, _annotation_ refers to a Dart metadata
+  annotation (as opposed to, say, a type annotation).  A metadata
+  annotation begins with the character `@`, followed by either a reference
+  to a compile-time constant (such as [`Component`](#component)) or a call
+  to a constant constructor. See the [Dart Language Guide][metadata] for
```
Comment (kwalrath) [inline src/angular/glossary.md]: Guide -> tour

File: src/angular/glossary.md [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+<div id="Z" class="l-main-section"></div>
+
+## Zone
+<div class="l-sub-section" markdown="1">
+  Zones are a mechanism for encapsulating and intercepting
+  a Dart application's asynchronous activity.
+
+  Read more about zones in [this article][zones].
+
+  [zones]: https://www.dartlang.org/articles/libraries/zones</div>
```
Comment (kwalrath) [inline src/angular/glossary.md]: Clicking this link gave me a 404. I think the `</div>` messed up the URL definition.

The link went here: `https://www.dartlang.org/articles/libraries/zones%3C/div%3E`

File: src/angular/glossary.md (hunk 1 above)
Comment (chalin) [inline src/angular/glossary.md]: Fixed

File: src/angular/glossary.md (hunk 2 above)
Comment (chalin) [inline src/angular/glossary.md]: Good catch. Fixed. I was counting on the link checker to catch those, but I had forgotten that it only checks internal links by default. I'll improve on that in a followup PR.

**Comment by chalin** [comment]
@kwalrath - FYI, I added an extra commit to fix a problem that arises when scrolling to the start of a letter's section A-Z. Previously, the scroll position was misaligned.

**Pull request merged**